src/ast/nodes/shared/knownGlobals.ts

```diff
@@ -254,6 +254,7 @@ const knownGlobals: GlobalDescription = {
 		keys: PF,
 		fromEntries: O,
 		entries: PF,
+		values: PF,
```

Comment:
I'm not sure about this one. `Object.values` can have a side effect if the passed object has a side effectful getter property. But this is same with `Object.entries`.

The output of
```js
const foo = {
    get foo() {
        console.log('sideeffect')
        return 'foo'
    }
}
console.log('---keys---')
Object.keys(foo)
console.log('---values---')
Object.values(foo)
console.log('---entries---')
Object.entries(foo)
```
is
```
---keys---
---values---
sideeffect
---entries---
sideeffect
```

Should I change `entries` and `values` to `O` and add a test for this case?

Reply:
> Should I change `entries` and `values` to `O` and add a test for this case?

IMO, yes!

Reply:
+1 Yes please! I hope at some point we will be able to detect iterables by tracking the `Symbol.iterator` and having good default behavior for built-in iterators. Util then, we should always be conservative.

Reply:
Actually, you should be able to do it properly by defining a specific call effect handler like this:

```js
const GETTER_ACCESS: ValueDescription = {
	deoptimizeArgumentsOnCall: doNothing,
	getLiteralValue: getTruthyLiteralValue,
	hasEffectsWhenCalled({ args }, context) {
		const [_thisArgument, firstArgument] = args;
		return (
			!(firstArgument instanceof ExpressionEntity) ||
			firstArgument.hasEffectsOnInteractionAtPath(
				UNKNOWN_PATH,
				NODE_INTERACTION_UNKNOWN_CALL,
				context
			)
		);
	}
}
```

That should actually check for getters with side effects.

Reply:
Ah, this does not work because for performance reasons, it does not check all keys when tracking unknown paths. In that case, just use `O`.

Reply:
Actually on second look, maybe it should work. It should be this method https://github.com/rollup/rollup/blob/dc89523737db3021961aac66a71353c2c68fddd5/src/ast/nodes/shared/ObjectEntity.ts#L285-L334 but I did not really debug into it. I just did a test in the local REPL where it did not work. But on second look, it actually DOES check all accessors...

Reply:
I debugged it a bit, `this.getMemberExpression(key)` is returning `UNKNOWN_EXPRESSION`, so the return of calling `GETTER_ACCESS.hasEffectsWhenCalled` function always is ture.

Reply:
Thank you!
While debugging around that function, I noticed that `NODE_INTERACTION_UNKNOWN_CALL` might should be  `NODE_INTERACTION_UNKNOWN_ACCESS`.
```js
const GETTER_ACCESS: ValueDescription = {
	deoptimizeArgumentsOnCall: doNothing,
	getLiteralValue: getTruthyLiteralValue,
	hasEffectsWhenCalled({ args }, context) {
		const [_thisArgument, firstArgument] = args;
		return (
			!(firstArgument instanceof ExpressionEntity) ||
			firstArgument.hasEffectsOnInteractionAtPath(
				UNKNOWN_PATH,
				NODE_INTERACTION_UNKNOWN_CALL, // maybe this should be NODE_INTERACTION_UNKNOWN_ACCESS?
				context
			)
		);
	}
}
```
After I changed this, it worked as I expect. I pushed that change.